pyqt-apps/siriushla/common/afc_board/utils.py

```diff
@@ -0,0 +1,396 @@
+
+        elif device.sub.endswith(('SA', 'SB', 'SP')):
+            return 6
+
+        elif device.sub.endswith(('M1', 'M2')):
+            return 7
+
+        elif device.sub.endswith(('C1')):
+            return 8
+
+        elif device.sub.endswith(('C2')) or device.sub.endswith(('C3')) and device.idx == '1':
+            return 9
+
+        elif device.sub.endswith(('C4')) or device.sub.endswith(('C3')) and device.idx == '2':
+            return 10
+
+    elif device.sec == 'TB':
+        if device.sub == '01':
+            return 6
+
+        elif device.sub == '02':
+            return 7
+
+        elif device.sub in ('03', '04'):
+            return 8
+
+    elif device.sec == 'TS':
+        if device.sub in ('01', '02'):
+            return 9
+
+        elif device.sub == '03' or device.sub == '04' and device.idx == '1':
+            return 10
+
+        elif device.sub == '04' and device.idx == '2':
+            return 11
+
+    elif device.sec == 'BO':
+        amc_bo = int(device.sub[:2])
+
+        if amc_bo in (BO_N + BO_N1):
+            return 11
+
+        elif amc_bo in BO_N2:
+            return 12
+
+
+def device2crate(device):
+    device = SiriusPVName(device)
+    if device.sec in ('TS', 'TB'):
+        return 'IA-20RaBPMTL'
+
+    elif device.sec == 'BO':
+        bo_num = int(device.sub[:2])
+
+        if bo_num in BO_N:
+            area_index = BO_N.index(bo_num) + 1
+
+        elif bo_num in BO_N1:
+            area_index = BO_N1.index(bo_num) + 1
+
+        elif bo_num in BO_N2:
+            area_index = BO_N2.index(bo_num) + 1
+
+        else:
+            return None
+
+        area_str = f"{area_index:02}"
+        return f'IA-{area_str}RaBPM'
+
+    elif device.sec == 'IA':
+        return f'IA-{device.sub}'
+
+    else:
+        return f'IA-{device.sub[:2]}RaBPM'
+
+
+class AFCAdvancedSettings(BaseWidget):
+    def __init__(self, parent=None, prefix='', display='', **kwargs):
+        super().__init__(parent=parent, prefix=prefix, **kwargs)
+        self.display = display
+        if display.sec == 'SI':
+            self.setObjectName('SIApp')
+        elif display.sec == 'BO':
+            self.setObjectName('BOApp')
+        elif display.sec == 'TS':
+            self.setObjectName('TSApp')
+        elif display.sec == 'TB':
+            self.setObjectName('TBApp')
+        elif display.sec == 'IA':
+            if display.dev == 'AMCFPGAEVR':
+                self.setObjectName('ASApp')
+            elif display.dev == 'FOFBCtrl':
+                self.setObjectName('SIApp')
+        self.setupui()
+
+    def setupui(self):
+        lay = QVBoxLayout(self)
+        if self.display.dev == 'FOFBCtrl':
+            lab = QLabel(
+                f'<h2>{self.display} AFCv4.0 Settings</h2>', self,
+                alignment=Qt.AlignCenter)
+        else:
+            lab = QLabel(
+                f'<h2>{self.display} AFCv3.1 Settings</h2>', self,
+                alignment=Qt.AlignCenter)
+
```

Comment:
you can use the if statement only to define the string to be shown in the `QLabel` and create the `lab` variable outside the if statement